File: aura_backend/src/main/java/com/backend/aura/modules/wellness/model/WellnessComment.java
package com.backend.aura.modules.wellness.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "wellness_comments")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WellnessComment {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false)
    private String postId;

    @Column(nullable = false)
    private String userId;

    @Column(nullable = false, length = 1000)
    private String originalContent;

    @Column(length = 1000)
    private String translatedContent;

    private String detectedLanguage;

    @Builder.Default
    @Enumerated(EnumType.STRING)
    private TranslationStatus translationStatus = TranslationStatus.PENDING;

    @Builder.Default
    private boolean isApproved = true;

    @Builder.Default
    private boolean isHidden = false;

    @Builder.Default
    private LocalDateTime createdAt = LocalDateTime.now();

    private LocalDateTime updatedAt;

    private String moderatedBy;

    private LocalDateTime moderatedAt;

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public enum TranslationStatus {
        PENDING,
        TRANSLATED,
        FAILED,
        NOT_NEEDED
    }
}

File: aura_app/lib/features/wellness/data/models/wellness_comment.dart
class WellnessComment {
  final String id;
  final String postId;
  final String userId;
  final String? userName;
  final String? userProfileImage;
  final String content;
  final String? translatedContent;
  final String? detectedLanguage;
  final String translationStatus;
  final DateTime createdAt;

  WellnessComment({
    required this.id,
    required this.postId,
    required this.userId,
    this.userName,
    this.userProfileImage,
    required this.content,
    this.translatedContent,
    this.detectedLanguage,
    this.translationStatus = 'PENDING',
    required this.createdAt,
  });

  bool get isEnglish =>
      detectedLanguage?.toLowerCase() == 'en' ||
      translationStatus == 'NOT_NEEDED';

  bool get canTranslate => translationStatus == 'PENDING' && !isEnglish;

  bool get hasTranslation =>
      translatedContent != null && translationStatus == 'TRANSLATED';

  factory WellnessComment.fromJson(Map<String, dynamic> json) {
    return WellnessComment(
      id: json['id'] ?? '',
      postId: json['postId'] ?? '',
      userId: json['userId'] ?? '',
      userName: json['userName'],
      userProfileImage: json['userProfileImage'],
      content: json['content'] ?? '',
      translatedContent: json['translatedContent'],
      detectedLanguage: json['detectedLanguage'],
      translationStatus: json['translationStatus'] ?? 'PENDING',
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
    );
  }
}
